recipe/meta.yaml

```diff
@@ -244,7 +245,8 @@ outputs:
         - python
       run_constrained:
         - apache-arrow-proc =*={{ build_ext }}
-        - cudatoolkit >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
+        - cuda-version >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
+        - cudatoolkit >={{ cuda_compiler_version_min }}   # [cuda_compiler_version != "None"]
```

Comment:
```suggestion
        - __cuda >= 9   # [cuda_compiler_version != "None"]
```

Reply:
Let's discuss above ( https://github.com/conda-forge/arrow-cpp-feedstock/pull/1079#discussion_r1217603197 ) and follow up after that thread concludes

Reply:
AFAICT, these run-constraints are not relevant for `pyarrow` & `pyarrow-tests`, as they _already_ run-depend on `libarrow`, which brings the `__cuda` constraint. I'm removing these.